src/doc/rustc/src/lints/levels.md

```diff
@@ -33,6 +34,40 @@ But this code violates the `missing_docs` lint.
 These lints exist mostly to be manually turned on via configuration, as we'll
 talk about later in this section.
 
+## expect
+
+Sometimes, it can be helpful to suppress lints, but at the same time ensure that
+the code in question still emits them. The 'expect' level does exactly this. If
+the lint in question is not emitted, the `unfulfilled_lint_expectation` lint
+triggers on the `expect` attribute, notifying you that the expectation is no
+longer fulfilled.
+
+```rust
+fn main() {
+    #[expect(unused_variables)]
+    let unused = "Everyone ignores me";
+
+    #[expect(unused_variables)]
+    let used = "I'm useful";
```

Comment:
nit: maybe add a comment in the code as to make it more obvious which `#[expect]` is unfulfilled
```suggestion
    #[expect(unused_variables)] // unused_variables lint is not emitted
    let used = "I'm useful";    // the expectation is therefore unfulfilled
```